### ISSUE TITLE
RFC: create `useUnit` hook for React components

### ISSUE THREAD

**Issue by sergeysova** [effector/effector#680]
Now we have two hooks: `useStore` and `useEvent`. But multiple usages of `useStore` will update the component twice or more, if some stores updated simultaneously. `useEvent` hook can bind not only events as well as effects.

Also, there is the `useEvent` proposal in the React ecosystem: https://github.com/reactjs/rfcs/pull/220

## Proposal

Create single hook `useUnit` accepts any kind of units and add batching under the hood.

```ts
const value = useUnit($store)
const handler = useUnit(someHappened)
const loadData = useUnit(loadDataFx)

// also supports mixing them all
const { value, handler, loadData } = useUnit({
  value: $store,
  handler: someHappened,
  loadData: loadDataFx,
})
```

I think `useUnit` should supports array shape:

```ts
const [handler, data] = useUnit([someHappened, $store])
```

## Use case

First of all it is simplification of the React bindings API. Using just one hook instead of two for the three kind of units.

Another use case is batching under the hood.

```ts
const { foo, bar } = useUnit({
  foo: $foo,
  bar: $bar,
})
```

At the moment, `useEvent` allows us to pass many events and effects:

```ts
const { handleA, handleB } = useEvent({
  handleA: someHappenedA,
  handleB: someHappenedB,
})
```

## Questions

**But what if someone passed two+ arguments?** Looks like we need to require the user to pass array, object, or just one argument explicitly. Otherwise there is will be too much function overloads. For example, there is will be two overloads with the same result:

```ts
const [first, second] = useUnit(a, b)
const [first, second] = useUnit([a, b])
```

## Another changes

Looks like `useStoreMap` and `useList` cannot be merged into `useUnit`. It is different kind of hooks.

**Comment by AlexandrHoroshih** [comment]
What if objects with shape `{ store, fn, keys }` will be treated as a `useStoreMap`?

```ts
const { foo } = useUnit({
  foo: { store: $foo, fn: (f, [a,b]) => f ?? null, keys: [a, b] },
})
```

**Comment by zerobias** [comment]
@AlexandrHoroshih it would be really cool if the types will allow that. Need to do a smoke test in the ts repl to prove that it will work in a first place

**Comment by sergeysova** [comment]
@AlexandrHoroshih I suggest to split your proposal as a separate RFC

**Comment by AlexandrHoroshih** [comment]
Released in `effector-react@22.1.0`
